core/unit_test/TestAtomicOperations.hpp

```diff
@@ -277,6 +277,63 @@ struct LoadStoreAtomicTest {
   static const char* name() { return "load/store"; }
 };
 
+#ifdef KOKKOS_ENABLE_DEPRECATED_CODE_4
+#ifdef KOKKOS_ENABLE_DEPRECATION_WARNINGS
+KOKKOS_IMPL_DISABLE_DEPRECATED_WARNINGS_PUSH()
+#endif
+struct DeprecatedAssignAtomicTest {
+  template <class T>
+  KOKKOS_FUNCTION static auto atomic_op(T* ptr_op, T* ptr_fetch_op,
+                                        T* ptr_op_fetch, T update) {
+    T old_val = Kokkos::atomic_load(ptr_op);
+    Kokkos::atomic_assign(ptr_op, update);
+    Kokkos::atomic_assign(ptr_op_fetch, update);
+    Kokkos::atomic_assign(ptr_fetch_op, update);
+    return Kokkos::pair<T, T>(old_val, update);
+  }
+  template <class T>
+  KOKKOS_FUNCTION static T op(T, T update) {
+    return update;
+  }
+  static const char* name() { return "load/assign"; }
+};
+
+struct DeprecatedIncrementAtomicTest {
```

Comment:
Are copy paste from `{Inc,Dec}AtomicTest` but with the atomic "reduce" operation that uses there deprecated version.  They are no fetch counterparts to test